### PR TITLE
Coverage upload on Windows - fixes #61

### LINKED ISSUE #61: Coverage report upload failing on appveyor with v2.2.0
Here is the command ran:

```
.\node_modules\.bin\istanbul cover .\node_modules\mocha\bin\_mocha  -- --reporter lcovonly --ui bdd -R spec -t 5000 tests\unit
```

The output:

```
=============================================================================
Writing coverage object [C:\projects\cross-browser-tests-runner\coverage\coverage.json]
Writing coverage reports at [C:\projects\cross-browser-tests-runner\coverage]
=============================================================================
=============================== Coverage summary ===============================
Statements   : 99.21% ( 501/505 )
Branches     : 89.83% ( 159/177 )
Functions    : 99.15% ( 116/117 )
Lines        : 99.21% ( 500/504 )
================================================================================
.\node_modules\.bin\codecov
  _____          _  
 / ____|        | |  
| |     ___   __| | ___  ___ _____   __  
| |    / _ \ / _` |/ _ \/ __/ _ \ \ / /  
| |___| (_) | (_| |  __/ (_| (_) \ V /  
 \_____\___/ \__,_|\___|\___\___/ \_/  
                                v2.2.0
==> Detecting CI Provider
    AppVeyor CI Detected
==> Configuration: 
    Endpoint: https://codecov.io
{ service: 'appveyor',
  commit: '9f8c9ac60e2e08491d5a6a533335bbfebb76cf1b',
  branch: 'master',
  pr: undefined,
  job: 'reeteshranjan/cross-browser-tests-runner/1.0.47',
  build: '8dkkhy310n3fteu4',
  slug: 'reeteshranjan/cross-browser-tests-runner',
  token: '6ddf3961-0a50-4816-bf4c-e0cfb5f3a922',
  package: 'node-v2.2.0' }
==> Building file structure
==> Generating gcov reports (skip via --disable=gcov)
    $ find . -type f -name '*.gcno'  -exec gcov  {} +
File not found - '*.gcno'
    Failed to run gcov command.
==> Scanning for reports
FIND: Parameter format not correct
child_process.js:524
    throw err;
    ^
Error: Command failed: find . -type f \( -name '*coverage.*' -or -name 'nosetests.xml' -or -name 'jacoco*.xml' -or -name 'clover.xml' -or -name 'report.xml' -or -name 'cobertura.xml' -or -name 'luacov.report.out' -or -name 'lcov.info' -or -name '*.lcov' -or -name 'gcov.info' -or -name '*.gcov' -or -name '*.lst' \) -not -name '*.sh' -not -name '*.data' -not -name '*.py' -not -name '*.class' -not -name '*.xcconfig' -not -name 'Coverage.profdata' -not -name 'phpunit-code-coverage.xml' -not -name 'coverage.serialized' -not -name '*.pyc' -not -name '*.cfg' -not -name '*.egg' -not -name '*.whl' -not -name '*.html' -not -name '*.js' -not -name '*.cpp' -not -name 'coverage.jade' -not -name 'include.lst' -not -name 'inputFiles.lst' -not -name 'createdFiles.lst' -not -name 'coverage.html' -not -name 'scoverage.measurements.*' -not -name 'test_*_coverage.txt' -not -path '*/vendor/*' -not -path '*/htmlcov/*' -not -path '*/home/cainus/*' -not -path '*/virtualenv/*' -not -path '*/js/generated/coverage/*' -not -path '*/.virtualenv/*' -not -path '*/virtualenvs/*' -not -path '*/.virtualenvs/*' -not -path '*/.env/*' -not -path '*/.envs/*' -not -path '*/env/*' -not -path '*/envs/*' -not -path '*/.venv/*' -not -path '*/.venvs/*' -not -path '*/venv/*' -not -path '*/venvs/*' -not -path '*/.git/*' -not -path '*/.hg/*' -not -path '*/.tox/*' -not -path '*/__pycache__/*' -not -path '*/.egg-info*' -not -path '*/$bower_components/*' -not -path '*/node_modules/*' -not -path '*/conftest_*.c.gcov'
FIND: Parameter format not correct
    at checkExecSyncError (child_process.js:481:13)
    at execSync (child_process.js:521:13)
    at Object.upload (C:\projects\cross-browser-tests-runner\node_modules\codecov\lib\codecov.js:264:18)
    at Object.<anonymous> (C:\projects\cross-browser-tests-runner\node_modules\codecov\bin\codecov:26:9)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
Command exited with code 1
```

What am I doing wrong here?

Linux build on circlci and OSX build on travis have been working fine for last one week or so across several builds already created.

The appveyor build url: https://ci.appveyor.com/project/reeteshranjan/cross-browser-tests-runner/build/1.0.47

appveyor.yml url: https://github.com/reeteshranjan/cross-browser-tests-runner/blob/master/appveyor.yml

### PR DESCRIPTION
**Pull request by reeteshranjan**
Uploading coverage data from Windows machines works with these changes. The changes have been tested using Appveyor.

### Modifications

- Ported the coverage upload code to work on Windows
   - Paths and files exclusion and inclusion filters
   - Exclusions from `bowerrc`
   - Exclusions from `gcov-global`
   - Executing `gcov` on `*.gcno`
   - Creating upload text

### Testing

Please see this sample project: https://github.com/reeteshranjan/codecov-node-appveyor-sample. It uses the modified version of `lib/codecov.js` (as in this commit) patched during the build to the local installation of codecov 2.2.0 npm module. Details are as follows:

- Multiple directories and files were created under `win-tests` (in the sample project) to test all the inclusion and exclusion filters.
- A sample `.bowerrc` file is added to test the port of handling exclusions as per `.bowerrc`.
- Sample `C` files were added, which are compiled with `--coverage` and run in the Appveyor build (link provided below). The porting of running `gcov` on the `*.gcno` files created on Windows was verified to include the resultant coverage of these `C` files in codecov.io reports (link provided below).
- Handling exclusions as per `gcov-global` option was verified locally.
- All `nodejs` versions supported by `codecov-node` have been included in the Appveyor build. As per https://www.appveyor.com/docs/lang/nodejs-iojs/#selecting-nodejs-or-iojs-version, the versions included are 7, 6, 5, 4, 0.12 and 1.0 (for iojs one needs to specify 1.0).
- As per http://help.appveyor.com/discussions/questions/372-build-setup-for-a-c-program, MinGW `gcc` is available on Appveyor, and the `appveyor.yml` in this sample project sets the path for `gcov` accordingly. However; the code stays with `gcov` being available on path, or as specified through command line. The `appveyor.yml` in the sample project can be referred to for other users to help them get their builds running on Appveyor.

Appveyor build: https://ci.appveyor.com/project/reeteshranjan/codecov-node-appveyor-sample/build/1.0.8
Successful coverage reports: https://codecov.io/gh/reeteshranjan/codecov-node-appveyor-sample

### CODE REVIEW

**Comment by reeteshranjan** [comment]
I can see the code coverage going down; however, this can be fixed if we had an Appveyor build. I am not sure how I could help on getting one. Please let me know if I can, and how, and I will.

**Comment by reeteshranjan** [comment]
I have setup an appveyor build for my fork. The latest build is at: https://ci.appveyor.com/project/reeteshranjan/codecov-node/build/1.0.2, which is successful. It uses `appveyor.yml` equivalent to existing `.travis.yml` in the sense that it supports all `nodejs` versions, runs `npm run test` and performs the last step of uploading `coverage.json`. The build is passing.

**Comment by stevepeak** [comment]
@eddiemoore How does this look?

File: lib/codecov.js [hunk 1]
```diff
@@ -238,11 +316,20 @@ var upload = function(args, on_success, on_failure){
   }
 
   // Detect .bowerrc
-  var bowerrc = execSync('cd '+root+' && test -f .bowerrc && cat .bowerrc || echo ""').toString().trim(), more_patterns = '';
+  var bowerrc;
+  if(!isWindows) {
+    bowerrc = execSync('cd '+root+' && test -f .bowerrc && cat .bowerrc || echo ""').toString().trim(), more_patterns = '';
+  } else {
+    bowerrc = execSync('cd '+root+' && if exist .bowerrc type .bowerrc').toString().trim(), more_patterns = '';
```
Comment (eddiemoore) [inline lib/codecov.js]: Move `, more_patterns` to line 319 since it's meant to be defining it. same goes with the previous section.

Comment (reeteshranjan) [inline lib/codecov.js]: @eddiemoore 
The variable `more_patterns` is defined at line 11. Are you saying that I should move the definition from line 11 to line 319? Or is it more about removing the repetitive assignment of `more_patterns` with an empty string, by factoring it out to a common place? If so, that's also done on line 11. Please check and advise.

Comment (eddiemoore) [inline lib/codecov.js]: I think we can just remove from this section since it's already defined as an empty string on line 11

**Comment by reeteshranjan** [comment]
Done. The sample app build https://ci.appveyor.com/project/reeteshranjan/codecov-node-appveyor-sample/build/1.0.9 has passed as well.

File: lib/codecov.js (hunk 1 above)
Comment (reeteshranjan) [inline lib/codecov.js]: I have added a new commit for the above. Is there anything else you need from me?

**Pull request merged**